From 875c39be53ac090fd52c410081e99d2ea17873b9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 11 Jun 2014 12:05:20 +0800
Subject: [PATCH] ARM: rockchip: rk3288: power down GPU domain by default

vdd_gpu may power off when boot, but pmu do not know.
So make pmu power off GPU power domain by default.
---
 arch/arm/mach-rockchip/rk3288.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rockchip/rk3288.c b/arch/arm/mach-rockchip/rk3288.c
index 8232c3ce494c..5d26864e2885 100644
--- a/arch/arm/mach-rockchip/rk3288.c
+++ b/arch/arm/mach-rockchip/rk3288.c
@@ -157,6 +157,10 @@ static void __init rk3288_dt_map_io(void)
 	writel_relaxed(24, RK_PMU_VIRT + RK3288_PMU_GPU_PWRDWN_CNT);
 	writel_relaxed(24, RK_PMU_VIRT + RK3288_PMU_GPU_PWRUP_CNT);
 
+	/* power down GPU domain by default */
+	v = readl_relaxed(RK_PMU_VIRT + RK3288_PMU_PWRDN_CON);
+	writel_relaxed(v | BIT(9), RK_PMU_VIRT + RK3288_PMU_PWRDN_CON);
+
 	rk3288_boot_mode_init();
 }
 
-- 
2.35.3

